x-pack/plugins/observability_solution/investigate_app/server/clients/create_entities_es_client.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ESSearchRequest, InferSearchResponseOf } from '@kbn/es-types';
+import type { KibanaRequest } from '@kbn/core/server';
+import { ElasticsearchClient } from '@kbn/core/server';
+import { entitiesAliasPattern, ENTITY_LATEST, ENTITY_HISTORY } from '@kbn/entities-schema';
+import { unwrapEsResponse } from '@kbn/observability-plugin/common/utils/unwrap_es_response';
+import {
+  MsearchMultisearchBody,
+  MsearchMultisearchHeader,
+} from '@elastic/elasticsearch/lib/api/types';
+
+export const SERVICE_ENTITIES_LATEST_ALIAS = entitiesAliasPattern({
+  type: 'service',
+  dataset: ENTITY_LATEST,
+});
+export const SERVICE_ENTITIES_HISTORY_ALIAS = entitiesAliasPattern({
+  type: 'service',
+  dataset: ENTITY_HISTORY,
+});
+export const HOST_ENTITIES_LATEST_ALIAS = entitiesAliasPattern({
+  type: 'host',
+  dataset: ENTITY_LATEST,
+});
+export const HOST_ENTITIES_HISTORY_ALIAS = entitiesAliasPattern({
+  type: 'host',
+  dataset: ENTITY_HISTORY,
+});
+export const CONTAINER_ENTITIES_LATEST_ALIAS = entitiesAliasPattern({
+  type: 'container',
+  dataset: ENTITY_LATEST,
+});
+export const CONTAINER_ENTITIES_HISTORY_ALIAS = entitiesAliasPattern({
+  type: 'container',
+  dataset: ENTITY_HISTORY,
+});
+type LatestAlias =
+  | typeof SERVICE_ENTITIES_LATEST_ALIAS
+  | typeof HOST_ENTITIES_LATEST_ALIAS
+  | typeof CONTAINER_ENTITIES_LATEST_ALIAS;
+
+type HistoryAlias =
+  | typeof SERVICE_ENTITIES_HISTORY_ALIAS
+  | typeof HOST_ENTITIES_HISTORY_ALIAS
+  | typeof CONTAINER_ENTITIES_HISTORY_ALIAS;
+
+export function cancelEsRequestOnAbort<T extends Promise<any>>(
+  promise: T,
+  request: KibanaRequest,
+  controller: AbortController
+): T {
+  const subscription = request.events.aborted$.subscribe(() => {
+    controller.abort();
+  });
+
+  return promise.finally(() => subscription.unsubscribe()) as T;
+}
+
+export interface EntitiesESClient {
+  search<TDocument = unknown, TSearchRequest extends ESSearchRequest = ESSearchRequest>(
+    indexName: string,
+    searchRequest: TSearchRequest
+  ): Promise<InferSearchResponseOf<TDocument, TSearchRequest>>;
+  msearch<TDocument = unknown, TSearchRequest extends ESSearchRequest = ESSearchRequest>(
+    allSearches: TSearchRequest[]
+  ): Promise<{ responses: Array<InferSearchResponseOf<TDocument, TSearchRequest>> }>;
+}
+
+export function createEntitiesESClient({
```

Comment:
Can't you use https://github.com/elastic/kibana/blob/main/x-pack/packages/observability/observability_utils/es/client/create_observability_es_client.ts#L32?

Reply:
This is a client specifically for searching through entities indices, but I should be using the observability es client as a dependency. Will update when I can.